packages/noya-utils/src/getIncrementedName.ts

```diff
@@ -0,0 +1,19 @@
+const numberSuffixRegExp = /(.*?)(\s\d+)?$/;
+
+export function getIncrementedName(
+  originalName: string,
+  names: string[],
+): string {
+  const [, prefix] = originalName.match(numberSuffixRegExp) || [];
+
+  const numbers = [originalName, ...names]
+    .filter((name) => name.startsWith(prefix))
+    .map((name) => {
+      const [, , number] = name.match(numberSuffixRegExp) || [];
+      return parseInt(number || '1');
```

Comment:
This should work but I think it'd be clearer as `return number ? parseInt(number) : 1`, since there isn't really a need to parse '1' as a string